packages/material-ui/src/ClickAwayListener/ClickAwayListener.tsx

```diff
@@ -76,12 +76,19 @@ function ClickAwayListener(props: ClickAwayListenerProps): JSX.Element {
   const syntheticEventRef = React.useRef(false);
 
   React.useEffect(() => {
-    // Ensure that this component is not "activated" synchronously.
-    // https://github.com/facebook/react/issues/20074
-    setTimeout(() => {
+    const doc = ownerDocument(nodeRef.current);
+    function activate() {
       activatedRef.current = true;
-    }, 0);
+      doc.removeEventListener('mousedown', activate, { capture: true });
+      doc.removeEventListener('touchstart', activate, { capture: true });
+    }
+
+    doc.addEventListener('mousedown', activate, { capture: true, passive: true });
+    doc.addEventListener('touchstart', activate, { capture: true, passive: true });
+
     return () => {
+      doc.removeEventListener('mousedown', activate, { capture: true });
+      doc.removeEventListener('touchstart', activate, { capture: true });
```

Comment:
We are using different options. Will this impact the `removeEventListener`? Regarding the `passive` I guess we need it for the same reason as #22269 👍

Reply:
Same question for me. I think TS complained but that might be back when I used `once`. Will double check.

Reply:
TS complains about this because by spec event listeners are only matched by type, listener and capture option. So technically you don't need to specify `passive`. 

Though practically we still might want to do that in case of browser inconsistencies. MDN recommends this as well:

> It's worth noting that some browser releases have been inconsistent on this, and unless you have specific reasons otherwise, it's probably wise to use the same values used for the call to addEventListener() when calling removeEventListener().

-- https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#matching_event_listeners_for_removal